app/scripts/HiGlassComponent.js

```diff
@@ -2911,6 +2911,52 @@ class HiGlassComponent extends React.Component {
       track.removeViewportChanged = trackId =>
         this.removeScalesChangedListener(fromView, trackId);
       track.setDomainsCallback = (xDomain, yDomain) => {
+        if (!fromView) {
+          // If there is no `fromView`, then there must be a `projectionXDomain` instead.
+          // Update the viewconfig to reflect the new `projectionXDomain` array
+          // on the `viewport-projection-horizontal` track.
+          const { viewConfig } = this.state;
+          const newViewConfig = viewConfig;
+          for (const [viewI, view] of viewConfig.views.entries()) {
+            if (view.uid === viewUid) {
+              for (const [tracksPos, tracks] of Object.entries(view.tracks)) {
+                for (const [otherTrackI, otherTrack] of tracks.entries()) {
+                  if (otherTrack.uid === track.uid) {
+                    switch (track.type) {
+                      case 'viewport-projection-horizontal':
+                        newViewConfig.views[viewI].tracks[tracksPos][
+                          otherTrackI
+                        ].projectionXDomain = xDomain;
+                        break;
+                      case 'viewport-projection-vertical':
+                        newViewConfig.views[viewI].tracks[tracksPos][
+                          otherTrackI
+                        ].projectionYDomain = yDomain;
+                        break;
+                      case 'viewport-projection-center':
+                        newViewConfig.views[viewI].tracks[tracksPos][
+                          otherTrackI
+                        ].projectionXDomain = xDomain;
+                        newViewConfig.views[viewI].tracks[tracksPos][
+                          otherTrackI
+                        ].projectionYDomain = yDomain;
+                        break;
+                      default:
+                        console.warn(
+                          'Unexpected track type in setDomainsCallback'
+                        );
+                    }
+                  }
+                }
+              }
+            }
+          }
+
+          const newViews = this.processViewConfig(newViewConfig);
+          this.setState({ views: newViews, viewConfig: newViewConfig });
```

Comment:
I'm still not convinced that this is the right way to go about this. The viewport track's projection*Scales can change very rapidly so they shouldn't be part of the state. Why not store them in a separate variable e.g. this.viewportScales[viewportTrack.uid] and only add them to the viewconf on export? 

See `this.xScales` in `HiGlassComponent` for an example of how we accomplish this for locked scales in tracks.

Here's the key part where that information is added to viewconf on export: https://github.com/higlass/higlass/blob/90d2b29d09a9b1e8e6d79b205a0bb42c0803d802/app/scripts/HiGlassComponent.js#L3112

Reply:
That does seem like a better approach. In 2a7d3f1 I have added `this.projectionXDomains` and `this.projectionYDomains` to store the changing scale domain information. 

I did also add a call to `this.triggerViewChangeDb` because I do still want the update to trigger the `.on('viewConfig')` callback